src/lib/y2network/proposal_settings.rb

```diff
@@ -0,0 +1,121 @@
+# encoding: utf-8
+#
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2packager/package"
+
+module Y2Network
+  # Class that stores the proposal settings for network during installation.
+  class ProposalSettings
+    include Yast::Logger
+    include Yast::I18n
+
+    # @return [Boolean] network service to be used after the installation
+    attr_accessor :backend
+
+    # Constructor
+    def initialize
+      Yast.import "ProductFeatures"
+      Yast.import "PackagesProposal"
+      Yast.import "Lan"
+
+      @backend = use_network_manager? ? :network_manager : :wicked
+    end
+
+    # Adds the NetworkManager package to the {Yast::PackagesProposal} and sets
+    # NetworkManager as the backend to be used
+    def enable_network_manager!
+      Yast::PackagesProposal.AddResolvables("network", :package, ["NetworkManager"])
+      Yast::PackagesProposal.RemoveResolvables("network", :package, ["wicked"])
+
+      log.info "Enabling NetworkManager"
+      self.backend = :network_manager
+    end
+
+    # Add the wicked package to the {Yast::PackagesProposal} and sets wicked
+    # as the backend to be used
+    def enable_wicked!
+      Yast::PackagesProposal.AddResolvables("network", :package, ["wicked"])
+      Yast::PackagesProposal.RemoveResolvables("network", :package, ["NetworkManager"])
+
+      log.info "Enabling Wicked"
+      self.backend = :wicked
+    end
+
+    # Convenience method to obtain whether the NetworkManager package is
+    # available or not.
+    #
+    # @return [Boolean] false if no package available, true otherwise
+    def network_manager_available?
```

Comment:
It could be a private method, isn't it?

Reply:
Well, I actually exposed it

https://github.com/yast/yast-network/pull/727/files#diff-b206be758f56a39bbb81b3b2aebd6580R57